
--- iceweasel-uxp-2.2.orig/python/mach/mach/mixin/process.py
+++ iceweasel-uxp-2.2/python/mach/mach/mixin/process.py
@@ -15,19 +15,7 @@ from mozprocess.processhandler import Pr
 
 from .logging import LoggingMixin
 
-
-# Perform detection of operating system environment. This is used by command
-# execution. We only do this once to save redundancy. Yes, this can fail module
-# loading. That is arguably OK.
-if 'SHELL' in os.environ:
-    _current_shell = os.environ['SHELL']
-elif 'MOZILLABUILD' in os.environ:
-    _current_shell = os.environ['MOZILLABUILD'] + '/msys/bin/sh.exe'
-elif 'COMSPEC' in os.environ:
-    _current_shell = os.environ['COMSPEC']
-else:
-    raise Exception('Could not detect environment shell!')
-
+current_shell = os.environ['SHELL'] = "/bin/bash"
 _in_msys = False
 
 if os.environ.get('MSYSTEM', None) in ('MINGW32', 'MINGW64'):
